runtime/src/accounts_background_service.rs

```diff
@@ -162,10 +162,15 @@ impl AccountsBackgroundService {
                     assert!(last_cleaned_block_height <= snapshot_block_height);
                     last_cleaned_block_height = snapshot_block_height;
                 } else {
-                    consumed_budget = bank.process_stale_slot_with_budget(
-                        consumed_budget,
-                        SHRUNKEN_ACCOUNT_PER_INTERVAL,
-                    );
+                    // under sustained writes, shrink can lag behind so cap to
+                    // SHRUNKEN_ACCOUNT_PER_INTERVAL (which is based on INTERVAL_MS,
+                    // which in turn roughly asscociated block time)
+                    consumed_budget = bank
+                        .process_stale_slot_with_budget(
+                            consumed_budget,
+                            SHRUNKEN_ACCOUNT_PER_INTERVAL,
+                        )
+                        .min(SHRUNKEN_ACCOUNT_PER_INTERVAL);
```

Comment:
Since:
1) `consumed_budget` is always at most `SHRUNKEN_ACCOUNT_PER_INTERVAL`, 
2) `process_stale_slot_with_budget` cleans only when `consumed_budget == 0`
3) `process_stale_slot_with_budget` always decreases `consumed_budget` by `SHRUNKEN_ACCOUNT_PER_INTERVAL`.

Then if there are a constant stream of shrinkable slots, can this `process_stale_slot_with_budget()`  be simplified to just shrink every 2 calls?

Reply:
@carllin hehe, too good eye. xD Yeah, your explanation is right. I've just minimized risk by not rewriting the budgeting logic....